back-end/service.py

```diff
@@ -95,50 +131,58 @@ def updateSecret(secret, data, user, repo):
         log('debug', 'Secret updated in `{}`'.format(repo))
         return True
     else:
-        log('warning', 'Secret could not be updated in `{}`'.format(repo))
+        raise ServerError('Secret could not be updated in `{}`'.format(repo), 412)
         log('critical', 'Execution may not work properly from here.')
```

Comment:
The `log` after the raised exception will not be reached.

Reply:
Cured and pushed.